pkg/core/blockchain.go

```diff
@@ -412,14 +424,17 @@ func (bc *Blockchain) storeBlock(block *Block) error {
 						}
 					}
 					account.Balances[prevTXOutput.AssetID] = account.Balances[prevTXOutput.AssetID][:balancesLen-1]
+					if err = bc.dao.PutAccountState(account); err != nil {
+						return err
+					}
```

Comment:
You can do it once after this `if` (at the moment you have duplicating code in both branches).